sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -1318,6 +1322,18 @@ def get_when_task(input_task_when, depended_conditions):
       pipeline_run['metadata']['annotations'] = pipeline_run['metadata'].setdefault('annotations', {})
       pipeline_run['metadata']['annotations'].update(self.pipeline_annotations)
 
+    if self.security_context:
+      pipeline_run['spec']['podTemplate'] = pipeline_run['spec'].get('podTemplate', {})
+      for key, value in self.security_context.to_dict().items():
+        if value is not None:
+          pipeline_run['spec']['podTemplate']['securityContext'] = \
+            pipeline_run['spec']['podTemplate'].setdefault('securityContext', {})
+          pipeline_run['spec']['podTemplate']['securityContext'][key] = value
+    if self.automount_service_account_token:
+      pipeline_run['spec']['podTemplate'] = pipeline_run['spec'].get('podTemplate', {})
+      pipeline_run['spec']['podTemplate']['automountServiceAccountToken'] = \
```

Comment:
any reason not use bool for `automountServiceAccountToken` at very beginning: TektonPipelineConf?

Reply:
ok I fixed the logics